src/parser/monk/mistweaver/modules/core/HotTrackerMW.tsx

```diff
@@ -1,23 +1,29 @@
 import SPELLS from 'common/SPELLS';
+import { Options } from 'parser/core/Analyzer';
+import Combatant from 'parser/core/Combatant';
+import { ApplyBuffEvent, RefreshBuffEvent } from 'parser/core/Events';
 import HotTracker from 'parser/shared/modules/HotTracker';
 
-const REM_BASE_DURATION = 20000;
-const ENV_BASE_DURATION = 6000;
-const EF_BASE_DURATION = 8000;
+const REM_BASE_DURATION: number = 20000;
+const ENV_BASE_DURATION: number = 6000;
+const EF_BASE_DURATION: number = 8000;
 
-const UPWELLING = 4000;
-const MISTWRAP = 1000;
-const TFT_REM_EXTRA_DURATION = 10000;
+const UPWELLING: number = 4000;
+const MISTWRAP: number = 1000;
+const TFT_REM_EXTRA_DURATION: number = 10000;
 
 class HotTrackerMW extends HotTracker {
 
-  constructor(...args) {
-    super(...args);
+  mistwrapActive: boolean = false;
+  upwellingActive: boolean = false;
+
+  constructor(args: Options) {
+    super(args);
     this.mistwrapActive = this.selectedCombatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id);
     this.upwellingActive = this.selectedCombatant.hasTalent(SPELLS.UPWELLING_TALENT.id);
   }
 
-  calculateMaxDuration(event){
+  calculateMaxDuration(event: ApplyBuffEvent & {selectedCombatant: Combatant}){
```

Comment:
i'm confused about this, what is the combatant doing attached to an event

Reply:
(leaving a reply so I get pinged when this is answered)

Reply:
This is because I manually add selectedCombatant to the event since I get errors trying to do `this.selectedCombatant` in this child class (for some reason it is undefined)